python/common/org/python/types/Int.java

```diff
@@ -929,4 +929,16 @@ public boolean __setattr_null(java.lang.String name, org.python.Object value) {
         }
         throw new org.python.exceptions.TypeError("'" + ndigits.typeName() + "' object cannot be interpreted as an integer");
     }
+
+
+    @org.python.Method(
+        __doc__ = "factorial(num)"
+    )
+    public org.python.Object __factorial__() {
+        long result = 1;
+        for (int i = 2; i <= this.value; i++) {
```

Comment:
In the _math.factorial_ function description in Python's documentation (https://docs.python.org/3/library/math.html) it seems like the function is supposed to raise a _ValueError_ if the value to run the function on is not integral or is negative. It seems like this check is missing here in your function.

Reply:
Well spotted! This is fixed now